ios/Classes/Modules/CrashReportingApi.m

```diff
@@ -19,10 +21,28 @@ - (void)sendJsonCrash:(NSString *)jsonCrash isHandled:(NSNumber *)isHandled erro
     NSDictionary *stackTrace = [NSJSONSerialization JSONObjectWithData:objectData
                                                                options:NSJSONReadingMutableContainers
                                                                  error:&jsonError];
-    SEL reportCrashWithStackTraceSEL = NSSelectorFromString(@"reportCrashWithStackTrace:handled:");
-    if ([[Instabug class] respondsToSelector:reportCrashWithStackTraceSEL]) {
-        [[Instabug class] performSelector:reportCrashWithStackTraceSEL withObject:stackTrace withObject:isHandled];
+    NSInteger value = [isHandled integerValue];
+
+    if (value == 1) {
+        [IBGCrashReporting cp_reportNonFatalCrashWithStackTrace:stackTrace level:IBGNonFatalLevelError groupingString:nil userAttributes:nil
+        ];
+    } else {
+        [IBGCrashReporting cp_reportFatalCrashWithStackTrace:stackTrace  ];
+    
     }
```

Comment:
We can better cast `isHandled` to `BOOL` and rename the `value` variable to something like `isNonFatal` for clarity, and a little bit of formatting. What do you think?

```suggestion
    BOOL isNonFatal = [isHandled boolValue];
    if (isNonFatal) {
        [IBGCrashReporting cp_reportNonFatalCrashWithStackTrace:stackTrace
                                                          level:IBGNonFatalLevelError
                                                 groupingString:nil
                                                 userAttributes:nil];
    } else {
        [IBGCrashReporting cp_reportFatalCrashWithStackTrace:stackTrace];
    }
```